evennia/typeclasses/models.py

```diff
@@ -733,3 +736,145 @@ def at_rename(self, oldname, newname):
+        """
+        try: return reverse('%s-detail' % self._meta.verbose_name.lower(), kwargs={'pk': self.pk, 'slug': slugify(self.name)})
+        except: return '#'
+        
+    def web_update_url(self):
+        """
+        Returns the URI path for a View that allows users to update this
+        object.
+        
+        ex. Oscar (Character) = '/characters/oscar/1/change/'
+        
+        For this to work, the developer must have defined a named view somewhere
+        in urls.py that follows the format 'modelname-action', so in this case
+        a named view of 'character-update' would be referenced by this method.
+        
+        ex.
+        url(r'characters/(?P<slug>[\w\d\-]+)/(?P<pk>[0-9]+)/change/$', CharUpdateView.as_view(), name='character-update')
+        
+        If no View has been created and defined in urls.py, returns an
+        HTML anchor.
+        
+        This method is naive and simply returns a path. Securing access to
+        the actual view and limiting who can modify objects is the developer's 
+        responsibility.
+        
+        Returns:
+            path (str): URI path to object update page, if defined.
+            
+        """
+        try: return reverse('%s-update' % self._meta.verbose_name.lower(), kwargs={'pk': self.pk, 'slug': slugify(self.name)})
+        except: return '#'
+        
+    def web_delete_url(self):
+        """
+        Returns the URI path for a View that allows users to delete this object.
+        
+        ex. Oscar (Character) = '/characters/oscar/1/delete/'
+        
+        For this to work, the developer must have defined a named view somewhere
+        in urls.py that follows the format 'modelname-action', so in this case
+        a named view of 'character-detail' would be referenced by this method.
+
+        ex.
+        url(r'characters/(?P<slug>[\w\d\-]+)/(?P<pk>[0-9]+)/delete/$', CharDeleteView.as_view(), name='character-delete')
+        
+        If no View has been created and defined in urls.py, returns an
+        HTML anchor.
+        
+        This method is naive and simply returns a path. Securing access to
+        the actual view and limiting who can delete this object is the developer's 
+        responsibility.
+        
+        Returns:
+            path (str): URI path to object deletion page, if defined.
+            
+        """
+        try: return reverse('%s-delete' % self._meta.verbose_name.lower(), kwargs={'pk': self.pk, 'slug': slugify(self.name)})
+        except: return '#'
+        
+    def get_absolute_url(self):
+        """
+        Django construct; used by Django Sites framework and within the Admin
+        panel for reverse linking to the object detail page. 
+        
+        https://docs.djangoproject.com/en/2.1/ref/models/instances/#get-absolute-url
+        
+        Returns:
+            path (str): URI path to object detail page, if defined.
+            
+        """
+        return self.web_detail_url()
```

Comment:
Name this method `web_get_absolute_url` for consistency. Since Django requires it on a specific form, you add that as an alias below: 
```python
get_absolute_url = web_get_absolute_url   # django requires this name
```